### PR TITLE
Better unit testing and improvements from integration testing - MOM-81 MOM-82

### PR DESCRIPTION
**Pull request by rcrichton**
@devcritter would you mind having a look through this for me?

### CODE REVIEW

File: src/mhd2xds.coffee [hunk 1]
```diff
@@ -22,6 +39,13 @@ exports.mhd1Metadata2Xds = (metadata) ->
     seconds = '0' + seconds
   now = "#{year}#{month}#{day}#{hours}#{minutes}#{seconds}"
 
+  if metadata.documentEntry.patientId.indexOf '&' is -1 and metadata.documentEntry.patientId.indexOf 'ZAF' isnt -1
+    sourceId = metadata.documentEntry.patientId.replace 'ZAF', 'ZAF&0.0.0&ISO'
```
Comment (hnnesv) [inline src/mhd2xds.coffee]: We may need to support more than just ZAF as passport numbers are also catered for. In theory any ISO 3166-1 alpha-3 code is supported but I think only a few countries are used in production (ZWE and MOZ I think). You'll have to check with @armageddon

Comment (hnnesv) [inline src/mhd2xds.coffee]: actually come to think of it - maybe you should use the `root` in the CDA id as the assigning auth and just use the extension as is for the identifier: `{escapedExt}^^^&{root}&ISO` where `escapedExt = extension.replace('^', '\S\')` (i.e. [hl7 escaped](http://www.hl7standards.com/blog/2006/11/02/hl7-escape-sequences/))

File: src/mhd2xds.coffee [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+
+  select = xpath.useNamespaces
+    'v3': 'urn:hl7-org:v3'
+
+  birthDate = select 'string(//v3:ClinicalDocument/v3:recordTarget/v3:patientRole/v3:patient/v3:birthTime/@value)', doc
+  birthDate = birthDate.replace '%', ''
+
+  authorId = select 'string(//v3:ClinicalDocument/v3:author/v3:assignedAuthor/v3:id/@extension)', doc
+  if authorId.length is 0 then authorId = 'unknown'
+  authorAssigningAuthorityId = select 'string(//v3:ClinicalDocument/v3:author/v3:assignedAuthor/v3:id/@root)', doc
```
Comment (hnnesv) [inline src/mhd2xds.coffee]: Would it be possible to add handling for multiple authors? Specifically for this there's a second author capturing the device information (software source):

```
<author>
  <time value="201501011300"/>
  <assignedAuthor>
    <id root="9a560d61-85f1-4d7b-8ee2-090d2900f836"/>
    <assignedAuthoringDevice>
      <code code="applicationCode" codeSystem="56877fb7-e3a9-4ad5-bfb5-64d48a045e83"/>
      <softwareName>software name</softwareName>
    </assignedAuthoringDevice>
  </assignedAuthor>
</author>
```

**Comment by hnnesv** [comment]
@rcrichton just some comments, but otherwise it looks good!

File: src/mhd2xds.coffee (hunk 1 above)
Comment (rcrichton) [inline src/mhd2xds.coffee]: On chatting on hipcaht we came up with the solution of encoding the as follow to be standards compiant: `#{patientId}^^^&#{patientIdType}-#{patientIdAssigningAuthority}-526ef9c3-6f18-420a-bc53-9b733920bc67&ISO`

File: src/mhd2xds.coffee (hunk 2 above)
Comment (rcrichton) [inline src/mhd2xds.coffee]: Ok I'l give it a go, do you know how it is possible to tell which author element it is (ie. provider or software)?

Comment (hnnesv) [inline src/mhd2xds.coffee]: Yeah: a provider would have an `assignedPerson` element, while with the software it would be `assignedAuthoringDevice`.

**Comment by rcrichton** [comment]
Hey @devcritter I've addressed the commit, could you have a look again?

**Comment by hnnesv** [comment]
:+1:

**Pull request merged**